### DISCUSSION TITLE
Will Twitter scraping be affected by this API announcement?

### DISCUSSION THREAD

**Discussion by appleb4** [AAndyProgram/SCrawler#104] [General]
Seems they want to start charging for access

https://twitter.com/TwitterDev/status/1621026986784337922

**Comment by AAndyProgram** [comment]
Yeah, I saw that tweet. Actually, I don't know. We need to wait until February 9th to make sure SCrawler is ok. I guess if they don't change the API itself, Twitter can be downloadable through SCrawler. I don't know what they mean by `free access to the twitter API`. If they replace the current API, then no, SCrawler won't be able to download Twitter profiles. If there is currently a way to download Twitter data without authorization, then that's what they might mean. In this case, I think it won't affect SCrawler because SCrawler uses cookies and authorization data (tokens). So many "ifs", so I want to wait until the date.

**Reply by AAndyProgram** [reply]
On the other hand, if they do make their API payable, I won't even fix Twitter downloader in SCrawler. Musk is a fascist and he is leading the company to ruin. Many users have already left this network.

**Reply by appleb4** [reply]
> On the other hand, if they do make their API payable, I won't even fix Twitter downloader in SCrawler. Musk is a fascist and he is leading the company to ruin. Many users have already left this network.

Can't argue with that. It's a real shame to see how he leads such an important platform

**Reply by AAndyProgram** [reply]
Most of users who left Twitter switched to the [Mastodon](https://joinmastodon.org/) social network. This [site](https://mastodon.social/explore) is very similar to Twitter. What do you think?

**Reply by appleb4** [reply]
I've heard of it and it seems like the next safe choice should twitter get any less usable.

**Reply by AAndyProgram** [reply]
They changed the date. 🤦‍♂️ Now the date is February 13th: https://twitter.com/TwitterDev/status/1623467615539859456

**Reply by AAndyProgram** [reply]
Delayed again: https://twitter.com/TwitterDev/status/1625234161010343941

**Reply by AAndyProgram** [reply]
Still no news. On [February 18](https://twitter.com/TwitterDev/status/1626732269174943745), they announced that everything they said earlier is still relevant and they will roll-out these changes over the next few weeks. ☹️

**Reply by appleb4** [reply]
Gotta be some internal conflicts for how long it's taking them to clarify anything.
Do you think these will continue to work? https://gist.github.com/shobotch/5160017

**Reply by AAndyProgram** [reply]
>Gotta be some internal conflicts for how long it's taking them to clarify anything.

It seems to me that this is easier than internal conflicts. Users dissatisfaction is the main thing which stops them from deploying immediately. Over time, users will have time to get used to it, and their dissatisfaction will decrease. So they announce something and look at the reactions of users. If the reaction is too negative, they just delay again to reduce it.

>Do you think these will continue to work? https://gist.github.com/shobotch/5160017

IDK

**Reply by AAndyProgram** [reply]
https://alternativeto.net/news/2023/1/twitterrific-has-shut-down-following-twitter-s-ban-of-third-party-clients/

**Reply by AAndyProgram** [reply]
I don't really understand what it is about. Is it about posting new data via bots or about parsing too?
https://alternativeto.net/news/2023/3/twitter-announces-new-api-features-and-pricing-affecting-developers-and-researchers/

**Reply by AAndyProgram** [reply]
**The old Twitter bookmarks API has been deprecated.** I cannot retrieve **new** bookmarked posts though the old API. If anyone knows how to get bookmarks via the new API, please post a cURL request and I'll fix it in SCrawler. Otherwise, saved Twitter posts will be deprecated in the new version.

Old (currently used in SCrawler) API request:
`https://api.twitter.com/2/timeline/bookmark.json?screen_name=<UserName>&count=200&tweet_mode=extended&include_entities=true&include_user_entities=true&include_ext_media_availability=true`

[This](https://developer.twitter.com/en/docs/twitter-api/tweets/bookmarks/api-reference/get-users-id-bookmarks) doesn't work:
`https://api.twitter.com/2/users/:id/bookmarks`

**Reply by AAndyProgram** [reply]
https://github.com/AAndyProgram/SCrawler/issues/119

**Reply by appleb4** [reply]
Is it just me, or is Twitter not downloading anymore? Users show up as deleted for me. My log file is empty

**Reply by AAndyProgram** [reply]
Yes, unfortunately I faced this issue today. 😒 Since twitter is now a fascist network and bookmarks are downloading via gallery-dl, I'm thinking about changing my twitter algo to [gallery-dl](https://github.com/mikf/gallery-dl).

But the descriptions of the gallery-dl options are so truncated that I don't understand how to get all the tweets. We have two users download modes: 'User media only' and all the tweets (including retweets). So I can't figure out how to get retweets. If you know, please write here a command to get them.

**Reply by appleb4** [reply]
Good to know, thanks. Personally I would be okay without retweets, but I hope you can figure it out regardless.

Next release I'll test the freezing UI issue again, but I haven't seen it happen on the current latest release, could be because Twitter isn't downloading at all, but I also have scrawler running on a PC that I don't play games on so it's almost a non-issue at this point

**Reply by AAndyProgram** [reply]
So is the UI problem solved?

**Reply by appleb4** [reply]
More like something I can't test for while twitter isn't allowed to download. But if I'm the only one who's reported it consistently I'm willing to close the issue, thinking it's something on my end

**Reply by AAndyProgram** [reply]
>But if I'm the only one who's reported it consistently I'm willing to close the issue, thinking it's something on my end

Only if you want to. Personally, I don't used to close issues until the problem is gone. 😉

**Reply by AAndyProgram** [reply]
Solution found. I need a couple of days to test the new code.

**Reply by AAndyProgram** [reply]
Fixed. Please update to the [latest release](https://github.com/AAndyProgram/SCrawler/releases/latest).
